### PR TITLE
fix: clear stale leaves and inner nodes in `set_subtree`

### PR DESCRIPTION
**Pull request by sashass1315**
## What was wrong?

`set_subtree()`https://github.com/0xMiden/crypto/blob/bbf74938f4febb9144c11175d0dac8df27bafcfe/miden-crypto/src/merkle/smt/simple/mod.rs#L292 did not remove existing leaves and inner nodes in the insertion region before adding new subtree data. This caused stale data to remain in the tree, breaking the invariant `leaves.is_empty() == (root == EMPTY_ROOT)`https://github.com/0xMiden/crypto/blob/bbf74938f4febb9144c11175d0dac8df27bafcfe/miden-crypto/src/merkle/smt/simple/mod.rs#L186-L190 and causing `num_leaves()`https://github.com/0xMiden/crypto/blob/bbf74938f4febb9144c11175d0dac8df27bafcfe/miden-crypto/src/merkle/smt/simple/mod.rs#L144-L147 to return incorrect values. In debug mode, calling `is_empty()`https://github.com/0xMiden/crypto/blob/bbf74938f4febb9144c11175d0dac8df27bafcfe/miden-crypto/src/merkle/smt/simple/mod.rs#L186-L190 after inserting an empty subtree would panic.

## What changed?

Added cleanup logic to `set_subtree()` that removes all leaves and inner nodes within the insertion region before adding the new subtree's data. Also added tests to verify the fix.

### CODE REVIEW

File: miden-crypto/src/merkle/smt/simple/tests.rs [hunk 1]
```diff
@@ -452,6 +452,40 @@ fn test_simplesmt_set_subtree_entire_tree() {
...
+    assert_eq!(tree.num_leaves(), 5);
+
+    // Insert empty subtree of depth 1 at index 2 (covers leaves 4-5)
+    let empty_subtree = SimpleSmt::<1>::new().unwrap();
+    tree.set_subtree(2, empty_subtree).unwrap();
+
+    // Leaves 4 and 5 should be removed, leaves 0, 1, 7 should remain
+    assert_eq!(tree.num_leaves(), 3);
+    assert_eq!(tree.get_leaf(&LeafIndex::<3>::new(0).unwrap()), a);
+    assert_eq!(tree.get_leaf(&LeafIndex::<3>::new(1).unwrap()), b);
```
Comment (huitseeker) [inline miden-crypto/src/merkle/smt/simple/tests.rs]: The new tests cover empty subtree insertion well, but don't verify behavior when inserting a **non-empty** subtree over existing data. This is the main scenario the stale data fix targets. Could we add a test that inserts a non-empty subtree at depth > 1 into a populated tree and verifies the root matches a fresh build?

File: miden-crypto/src/merkle/smt/simple/mod.rs [hunk 2]
```diff
@@ -306,7 +306,7 @@ impl<const DEPTH: u8> SimpleSmt<DEPTH> {
         let subtree_root_index =
             NodeIndex::new(subtree_root_insertion_depth, subtree_insertion_index)?;
 
-        // add leaves
+        // remove existing leaves and inner nodes in the insertion region
```
Comment (huitseeker) [inline miden-crypto/src/merkle/smt/simple/mod.rs]: The CI changelog check is failing. Could you add an entry for this PR under the `## 0.23.0 (TBD)` section in CHANGELOG.md? Suggested entry:

- Fixed `SimpleSmt::set_subtree` to clear stale leaves and inner nodes in the insertion region.

**Comment by huitseeker** [comment]
Closed for inactiviy, in favor of #981
